### PR TITLE
Create GitHub Actions workflow for automatic publishing

### PR DESCRIPTION
**Pull request by acelaya**
Part of https://github.com/hypothesis/wp-hypothesis/issues/40

Closes https://github.com/hypothesis/wp-hypothesis/issues/39

Add a new GitHub actions workflow that triggers for every new release, and publishes the plugin to wordpress.org using https://github.com/10up/action-wordpress-plugin-deploy. This seems to be the de-facto GitHub Action to publish WordPress plugins.

Additionally, remove the custom scripts previously used for building and publishing.

### TODO

- [x] Make sure secrets are set

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,7 +13,19 @@ Stable versions are available on the
...
+2. **Create a tag** pointing at the version-change commit and generate a **new GitHub release** (details follow). Publishing a GitHub release will kick off a GitHub Action that will publish the plugin to wordpress.org
+
+### Creating a GitHub release
+
+Create a [new GitHub release](https://github.com/hypothesis/frontend-shared/releases/new/) with these values:
+
+1.  _Tag_: Create a new tag for the release, targeting the `main` branch (your just-merged version bump should be at the tip)[^2]. The tag should match the version number, e.g. `v5.2.1`.
+2.  _Title_: Use the tag name.
+3.  Click the `Auto-generate release notes` button to generate release notes and edit as needed. We use [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) formatting.[^3]
+4.  Leave other fields alone/as defaults.
```
Comment (acelaya) [inline README.md:28]: I took this from frontend-shared. The only difference is that the version needs to be edited in `hypothesis.php`.

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "hypothesis",
-  "version": "0.4.8",
+  "version": "0.6.0",
```
Comment (acelaya) [inline package.json:3]: This was outdated, but it's not really used at the moment.

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+      - name: Make language files
+        shell: bash
+        run: |
+          curl -O https://raw.githubusercontent.com/wp-cli/builds/gh-pages/phar/wp-cli.phar
+          php wp-cli.phar i18n make-pot . "languages/hypothesis.pot"
+      - name: Publish WordPress Plugin
+        uses: 10up/action-wordpress-plugin-deploy@stable
+        env:
+          SVN_USERNAME: ${{ secrets.WP_SVN_USERNAME }}
+          SVN_PASSWORD: ${{ secrets.WP_SVN_PASSWORD }}
```
Comment (acelaya) [inline .github/workflows/publish.yml:24]: The screts have been set

image

File: hypothesis.php [hunk 4]
```diff
@@ -4,7 +4,7 @@
  * Plugin URI: https://hypothes.is/
  * Description: Hypothesis is an open platform for the collaborative evaluation of knowledge. This plugin embeds the necessary scripts in your WordPress site to enable any user to use Hypothesis without installing any extensions.
  * Version: 0.6.0
- * Requires at least: 6.0
+ * Requires at least: 6.2
```
Comment (acelaya) [inline hypothesis.php:7]: Most popular plugins support the latest 3 minor versions, which are 6.2, 6.3 and 6.4

File: README.md [hunk 5]
```diff
@@ -14,7 +13,19 @@ Stable versions are available on the
...
```
Comment (acelaya) [inline README.md:28]: I took this from frontend-shared. The only difference is that the version needs to be edited in `hypothesis.php` and `readme.txt`.

File: README.md [hunk 6]
```diff
@@ -14,7 +13,19 @@ Stable versions are available on the
...
 
-New releases are automatically published to WordPress.org via Travis CI (using [this script](bin/deploy.sh)) whenever the version in [hypothesis.php](hypothesis.php) is incremented and a new version is tagged. Translation files will be updated as well. This method is largely based on the process described by [Iain Poulson](https://github.com/polevaultweb) in [this blog post](https://deliciousbrains.com/deploying-wordpress-plugins-travis/), and also makes use of the [wp-cli](https://wp-cli.org) `i18n` command.
+Follow these steps to publish a new plugin version.
+
+1. **Update the package version** in `hypothesis.php`, `readme.txt` and `package.json`, and merge that change into the `main` branch[^1]. We use [Semantic Versioning](https://semver.org/#semantic-versioning-200).
+2. **Create a tag** pointing at the version-change commit and generate a **new GitHub release** (details follow). Publishing a GitHub release will kick off a GitHub Action that will publish the plugin to wordpress.org
+
+### Creating a GitHub release
+
+Create a [new GitHub release](https://github.com/hypothesis/frontend-shared/releases/new/) with these values:
```
Comment (marcospri) [inline README.md:23]: This points to frontend-shared

**Review by marcospri — APPROVED** [review]
Not much too say, looks good :+1:

**Pull request merged**